core/src/main/kotlin/fr/sncf/osrd/envelope_sim_infra/MRSP.kt

```diff
@@ -72,36 +74,39 @@ fun computeMRSP(
     trainTag: String?,
     temporarySpeedLimitManager: TemporarySpeedLimitManager?,
     safetySpeedRanges: DistanceRangeMap<Speed>? = null,
+    ignoreInfraSpeedLimits: Boolean = false,
 ): Envelope {
     val builder = MRSPEnvelopeBuilder()
     val pathLength = toMeters(path.getLength())
 
     val offset = if (addRollingStockLength) rsLength else 0.0
     val speedLimitProperties = path.getSpeedLimitProperties(trainTag, temporarySpeedLimitManager)
-    for (speedLimitPropertyRange in speedLimitProperties) {
-        // Compute where this limit is active from and to
-        val start = toMeters(speedLimitPropertyRange.lower)
-        val end = min(pathLength, offset + toMeters(speedLimitPropertyRange.upper))
-        val speedLimitProp = rangeMapEntryToSpeedLimitProperty(speedLimitPropertyRange)
-        val speed = toMetersPerSecond(speedLimitProp.speed)
-        val attrs =
-            mutableListOf<SelfTypeHolder?>(
-                EnvelopeProfile.CONSTANT_SPEED,
-                MRSPEnvelopeBuilder.LimitKind.SPEED_LIMIT
-            )
-        if (speedLimitProp.source != null) {
-            attrs.add(speedLimitProp.source)
-        }
-        if (attrs.any { it is UnknownTag }) attrs.add(HasMissingSpeedTag)
-        if (speed != 0.0) {
-            // Add the envelope part corresponding to the restricted speed section
-            builder.addPart(
-                EnvelopePart.generateTimes(
-                    attrs,
-                    doubleArrayOf(start, end),
-                    doubleArrayOf(speed, speed)
+    if (!ignoreInfraSpeedLimits) { //if we take into account speedlimits coming from the infrastructure
```

Comment:
I'd have preferred a way to avoid this large `if` block.

Maybe something like this, what do you think?

```kotlin
    val speedLimitProperties =
        if (ignoreInfraSpeedLimits) distanceRangeMapOf()
        else path.getSpeedLimitProperties(trainTag, temporarySpeedLimitManager)
    for (speedLimitPropertyRange in speedLimitProperties) {
        // ...
    }
```